blog-posts/from-promises-to-observables/from-promises-to-observables.md

```diff
@@ -0,0 +1,342 @@
+                post.sourceURL = 'Medium';
+              } else if (post.sourceURL === 'https://rhythmandbinary.com/feed') {
+                post.sourceURL = 'Rhythm and Binary';
+              } else if (post.sourceURL === 'https://dev.to/feed/andrewevans0102') {
+                post.sourceURL = 'DEV.TO';
+              }
+              output.push(post);
+            })
+
+            output.sort((a: any, b: any) => {
+              return b.sortDate - a.sortDate;
+            });
+
+            return output;
+        }, []),
+        catchError(err => {
+          throw 'error in source observable. Message: ' + err.message;
+        }),
+        takeUntil(this.unsubscribe)
+      );
+  }
+```
+
+Here I am taking advantage of Angular's [HttpClient](https://angular.io/guide/http) which wraps HTTP calls in an observable.
+
+I then use the [merge](https://rxjs.dev/api/index/function/merge) operator to subscribe to all of the HttpClient calls and combine them into one output.
+
+The [scan](https://rxjs.dev/api/operators/scan) operator then takes the merged observables and appends the response to one common output.
+
+I include the [catchError](https://rxjs.dev/api/operators/catchError) operator to handle any errors in the stream, if one of the calls fails etc.
+
+I also use **pipe** to take the output of one observable and passes it into another. This is a common pattern with RxJS.
+
+The last operator that was passed into the **pipe** also references a [takeUntil](https://rxjs.dev/api/operators/takeUntil) operator. This is a very powerful RxJS operator that will unsubscribe an observable based on an event you pass in. Here I've created a [subject](https://rxjs.dev/guide/subject) which handles unsubscribing this main observable when the code finishes running. This is a fairly common pattern when handling observables. RxJS **subjects** also can be used for multicasting and doing observable like actions. I'm just using it here because it provides the behavior I wanted and makes a simple `unsubscribe` call clear out the resources. If I didn't do this it could cause `memory leaks` and potentially freeze up my browser session. You can see this behavior in the `reactive` component's `clear` method:
+
+```ts
+  clear() {
+    this.unsubscribe.next();
+    this.unsubscribe.complete();
+    this.posts$ = null;
+  }
+```
+
+Also note that I make the observable `null`. This is not necessary, but for the basic example application I wanted to visually show the data disappearing when `clear` was called.
+
+> As a disclaimer here, there are also 100+ other ways to do this same type of event handling. Additional implementations include using **from** to cast promises, **fromFetch** using the `fetch` API, or even an ajax implementation with `rxjsw`. I chose the implementation here because it was easy to follow and I was happy with the output. RxJS is very flexible and gives you the power to control the streams with (almost) any behavior.
```

Comment:
```suggestion
> As a disclaimer here, there are also 100+ other ways to do this same type of event handling. Additional implementations include using **from** to cast promises, **fromFetch** using the `fetch` API, or even an ajax implementation with `rxjs`. I chose the implementation here because it was easy to follow and I was happy with the output. RxJS is very flexible and gives you the power to control the streams with (almost) any behavior.
```